src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -81,7 +81,7 @@ public void storeMessage(Message message) {
     Query query = new Query("Message");
     PreparedQuery results = datastore.prepare(query);
     for (Entity entity : results.asIterable()) {
-      users.add((String)entity.getProperty("user"));
+      users.add(((String)entity.getProperty("user")).trim());
```

Comment:
What does this do?

Reply:
It eliminates leading and trailing spaces. Community.html was showing up duplicated usernames bc some of them contained trailing spaces.

Reply:
Hmm... I think the problem is at other code since it should not, in the first place, store a user with extra spaces.

Reply:
I think this is probably a problem exclusive to the local server. Because, when logging in on the deployed version, it uses Google login.

Reply:
http://su19-codeu-39-5571.appspot.com/community.html
There're 2 bbayarmaa now. I just deployed the version without trim. I'm trying to find why it stored username with extra space.

Reply:
@wed13 hey have you found the root issue? If you haven't you can discard the change in this line and get the pull request merged since the problem is not here.